### ISSUE TITLE
Shall we stick to icons or list the sounds by name?

### ISSUE THREAD

**Issue by felipeborges** [armonge/gnome-shell-extension-focusli#6]
The way it is now, with icons, kind of limits us to the system icons available or requires some extra work to download or ship icons within the extension (which doesn't scale if there will be dozens of sounds).

So **I would like to know whether someone else thinks we should go for a list of sounds names instead or stick to icons?**

**Comment by luisortizuriarte** [comment]
I like the icons, but creating a new icon seems like a lot more work than finding translations for all the sounds.

(Excellent idea for an extension btw!)

**Comment by Forage** [comment]
Names would solve the problem with custom added sounds at the same time, but I do think icons remain visually more appealing.
If the lack of better and more icons can not be solved then I'd vote for names since usability wins for me over appearance.

**Comment by julianrichen** [comment]
You could always do a mix of the two, a list which displays icons if they are available but always shows the sound name. Or you ship a generic sound icon as a default. Would also allow for more sounds while taking-up less space.

Example, each row is `[icon] [name] [slider]`:
```
                        /\
+-----------------------  ------------------------+
|                                                 |
|  [icon] Rain           |----------0----------|  |
|                                                 |
| ----------------------------------------------- |
|                                                 |
|  [icon] Wind           |----------0----------|  |
|                                                 |
| ----------------------------------------------- |
|                                                 |
|  [icon] Summer Night   |----------0----------|  |
|                                                 |
| ----------------------------------------------- |
|                                                 |
|  [icon] Fireplace      |----------0----------|  |
|                                                 |
| ----------------------------------------------- |
|                                                 |
|         Sound wo/ icon |----------0----------|  |
|                                                 |
+-------------------------------------------------+
```

**Comment by felipeborges** [comment]
@julianrichen very good point.

**Comment by lode** [comment]
For me the spacious interface, not only the icons, give the app a great feeling. I think having a default icon is better to keep the spacing.

**Comment by AbuMaia** [comment]
I second julianrichen's opinion.

**Comment by diazbastian** [comment]
The mix of icons + label is the best choice for the sounds included by default, also I would not change the appearance of the extension since most (all maybe) applications of this style have that appearance and is more familiar to users . Asoftmurmur is a good example worthy of imitation.

![asoftmurmur](https://cloud.githubusercontent.com/assets/5126124/26189408/8c2c95a2-3b72-11e7-95e4-f02b778d85f2.png)

On the other hand the interface for custom sounds must be different, like the list proposed above (without icons).

**Comment by felipeborges** [comment]
I'm glad that we had very relevant feedback. Thanks everyone! This way, I would like to ask a question within this question.

**If we would stick to icons, would someone be willing to draw them?**

**Comment by diazbastian** [comment]
@felipeborges Would focusli icons be based on GNOME symbolic icons?
That way you could reuse gnome-weather and other system icons, such as tv-symbolic for white noise, etc. So only a few icons to do.

I made a drawing (I draw badly, sorry) that represents the idea I mentioned above, it would only need 9 icons and 8 sounds by default, since others would be shown with another interface added from local files or the web search presented in another report. The play button is to solve the problem of starting several selected sounds and pausing them at the same time without having to do it individually (I added profiles, but I don't like the idea).

![focusliapp-mockup](https://cloud.githubusercontent.com/assets/5126124/26221215/b9d6493c-3be3-11e7-9632-6082824aada9.png)

The mockup I did it thinking about the suggestions I read here. I am of the idea that extensions should be kept simple, but if you add complexity the best model to follow is a mini app with a simple status indicator, like [gnome pomodoro](http://gnomepomodoro.org/) IHMO.

**Comment by felipeborges** [comment]
> @felipeborges Would focusli icons be based on GNOME symbolic icons?
> That way you could reuse gnome-weather and other system icons, such as tv-symbolic for white noise, > etc. So only a few icons to do.

@diazbastian we can support both symbolic and custom icons.

> The mockup I did it thinking about the suggestions I read here.

@diazbastian They are pretty good. You are officially the designer of the extension.

I agree entirely with your proposal with a single suggestion: I would place in the bottom of the popup two circled icons (to play and to open the settings dialog), just like the Shell's

![image](https://cloud.githubusercontent.com/assets/365434/26222580/da3d864a-3c1a-11e7-8baa-5933e9318a08.png)

What do you think?

**Comment by diazbastian** [comment]
@felipeborges Yeah, it's a good idea. I had put the buttons at the top to decrease the cursor stroke, but anyway it's fine.

> we can support both symbolic and custom icons.

I could see the panorama over the weekend and devote myself to create the missing icons (symbolic)

> They are pretty good. You are officially the designer of the extension.

I am not a designer, but one of the advantages of Free Software is the cooperation 👍 

--
I'm going to create a different report to discuss the mockups (#11) and keep this thread moving forward with the icons.
